### PR TITLE
Added code block functionality

### LINKED ISSUE #20: Image render does not support caption
There is currently breaking code where caption is declared and fetched by the API. It appears that Notion may have discontinued support for captions under images or are yet to include it.
[Notion API - Image Blocks](https://developers.notion.com/reference/block#image-blocks) Further down, captions are used for File Blocks and Bookmark Blocks.

I suggest commenting out/removing the use of captions within the Image block. Let me know if you would like me to contribute.

### PR DESCRIPTION
**Pull request by ggiande**
I added the bare minimum for a code block to render on the notion page. 

Requires: 
- Language Support
- Copying and Pasting the code block
- Polishing CSS

I am learning the basics and would appreciate any feedback/support. I am trying :) PR Currently holds the origin from my PR to fix #20.

### CODE REVIEW

File: pages/post.module.css [hunk 1]
```diff
@@ -53,6 +53,22 @@
   font-style: italic;
 }
 
+.pre_code_block {
+  background: rgb(15, 8, 28);
```
Comment (samuelkraft) [inline pages/post.module.css:57]: Can this block re-use styles from the `.code` class? It already has background styles that work with dark mode and sets to font to a monospaced one.

File: pages/post.module.css [hunk 2]
```diff
@@ -53,6 +53,22 @@
...
+  padding: 20px;
+  margin: 20px 0;
+  line-height: 2.3;
+  border-radius: 0.75em;
```
Comment (samuelkraft) [inline pages/post.module.css:61]: `border-radius: 12px;`  too keep consistency

File: pages/post.module.css [hunk 3]
```diff
@@ -53,6 +53,22 @@
   font-style: italic;
 }
 
+.pre_code_block {
```
Comment (samuelkraft) [inline pages/post.module.css:56]: Shorten to just `.pre`

**Review by samuelkraft — CHANGES_REQUESTED** [review]
Thanks for the PR! Just some minor comments then we're good to go :)

File: pages/post.module.css (hunk 2 above)
Comment (ggiande) [reply pages/post.module.css:61]: Gotcha.

File: pages/post.module.css (hunk 1 above)
Comment (ggiande) [reply pages/post.module.css:57]: Yes, however, padding will change, it looks a little tight with the current code padding.

File: pages/post.module.css (hunk 3 above)
Comment (ggiande) [reply pages/post.module.css:56]: Done

**Comment by samuelkraft** [comment]
Screenshot 2022-01-25 at 20 56 25
@ggiande awesome! Just one final thing, text isn't visible in light mode

**Comment by ggiande** [comment]
Unable to replicate issue on mac: image
Unable to replicate the issue on windows: 
image

**Comment by samuelkraft** [comment]
> Unable to replicate issue on mac: image
> Unable to replicate the issue on windows:
> image

The PR app above is outdated, I think it deployed your initial commit for some reason. Can you try running the latest commit locally?

**Comment by ggiande** [comment]
Pushed a new commit with the upstream changes, should be good now.

**Comment by samuelkraft** [comment]
> Pushed a new commit with the upstream changes, should be good now.

Can you fix the whitespace/indendation changes before merging?

**Comment by ggiande** [comment]
Whitespace/indentation fixed. Please check the changes.

**Review by samuelkraft — APPROVED** [review]

**Pull request merged**